### PR TITLE
molgw: add v3.3

### PR DESCRIPTION
**Pull request by bruneval**

### CODE REVIEW

File: var/spack/repos/builtin/packages/molgw/package.py [hunk 1]
```diff
@@ -18,22 +18,23 @@ class Molgw(MakefilePackage):
...
+    version("3.3", sha256="ff1c8eb736049e52608d4554a2d435ee9d15e47c4a9934d41712962748929e81")
     version("3.2", sha256="a3f9a99db52d95ce03bc3636b5999e6d92b503ec2f4afca33d030480c3e10242")
 
     variant("openmp", default=False, description="Build with OpenMP support")
     variant("scalapack", default=False, description="Build with ScaLAPACK support")
 
     depends_on("blas")
     depends_on("lapack")
-    depends_on("libxc@5:")
+    depends_on("libxc")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/molgw/package.py]: Why did we remove the minimum version?

File: var/spack/repos/builtin/packages/molgw/package.py [hunk 2]
```diff
@@ -50,7 +51,7 @@ class Molgw(MakefilePackage):
 
     def _get_mkl_ld_flags(self, spec):
         mklroot = str(getenv("MKLROOT"))
-        command = [mklroot + "/bin/intel64/mkl_link_tool", "-libs", "--quiet"]
+        command = [mklroot + "/bin/mkl_link_tool", "-libs", "--quiet"]
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/molgw/package.py:54]: Does this even work with OneAPI?

**Review by alalazo — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/molgw/package.py (hunk 1 above)
Comment (bruneval) [reply var/spack/repos/builtin/packages/molgw/package.py]: It's a mistake. Let me revert this in the next commit

File: var/spack/repos/builtin/packages/molgw/package.py (hunk 2 above)
Comment (bruneval) [reply var/spack/repos/builtin/packages/molgw/package.py:54]: Sure! it does.

Comment (alalazo) [reply var/spack/repos/builtin/packages/molgw/package.py:54]: Ok, I see the variable is set in the package.

**Review by alalazo — APPROVED** [review]

**Pull request merged**